samcli/commands/deploy/command.py

```diff
@@ -96,6 +99,14 @@
     "changes to be made to the stack. The default behavior is  to  return  a"
     "non-zero exit code.",
 )
+@click.option(
```

Comment:
I know we talked about this before, why is this explicitly required? can't we default to using JSON for the intermediate template?

Reply:
Sure, we can. user had the option with package, do we gain anything by removing it from deploy?

Reply:
There is no obvious user benefit of this flag in `sam deploy`. Why expose then?